sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -186,6 +187,9 @@ export class ApplicationService extends RecordDataModelService<Application> {
       application.studentAssessments = [originalAssessment];
       application.currentAssessment = originalAssessment;
       application.submittedDate = now;
+      application.location = {
+        id: applicationData.selectedLocation,
+      } as InstitutionLocation;
```

Comment:
Should we directly trust the form data and assign it to the location?? as a end user can always modify the submitted form data. In case of offerings we validate it with DB before assigning it to the application. 

![image](https://user-images.githubusercontent.com/54600590/232160712-1bca958d-9b0f-4a47-8463-1e45b7d07777.png)

Reply:
We can follow what @dheepak-aot suggested to avoid letting the DB throw the exception.

Reply:
Yes, I added the DB check for the location. Thanks!